code/code/misc/offense.cc

```diff
@@ -952,8 +952,8 @@ int TBeing::doFlee(const char *arg)
   for (i = 0; i < 20; i++) {
     dirTypeT attempt = dirTypeT(::number(MIN_DIR, MAX_DIR-1));        // Select a random direction 
     
-    // fighting, so give slight chance of letting PC direct the direction
-    if (chosenDir != DIR_NONE && !panic && !(::number(0,99) < (30+getSkillValue(skill)/2)))
+    // fighting, so give PC a skill-based chance to specify direction with a small failure chance at max skill
+    if (chosenDir != DIR_NONE && !panic && ::number(0,100) < 45+(getSkillValue(skill)/2))
```

Comment:
45+skillValue/2 seems arbitrary and doesn't use the methods in place for skill checks

Why not just call bsuccess() here? 

bool TBeing::bSuccess(spellNumT spell)

The rate of success is determined generally by the difficulty of the skill. Any easy task has a success chance of 90%. By using this method you can adjust the success chance by changing the difficulty in the spell info table and get things like hones for the skill without having to implement that yourself.

Reply:
This is just a guess, but is it possible they didn't use bSuccess there because it's inside a `for` loop that runs up to 20 times looking for possible flee directions, and every time bSuccess executes it potentially gives a skill hone?

I just remember seeing comments elsewhere in the code about not wanting to use bSuccess in a part of the code that gets called really frequently/repetitively and wondered if it was relevant here.

Reply:
```
for (i = 0; i < 20; i++) {
    dirTypeT attempt = dirTypeT(::number(MIN_DIR, MAX_DIR-1));        // Select a random direction 
    
    // fighting, so give slight chance of letting PC direct the direction
    if (chosenDir != DIR_NONE && !panic && !(::number(0,99) < (30+getSkillValue(skill)/2)))
      attempt = chosenDir;
    
    if (canFleeThisWay(this, attempt)) {
      if (panic || !doesKnowSkill(skill) || 
          !bSuccess(skill)) {
        act("$n panics, and attempts to flee.", TRUE, this, 0, 0, TO_ROOM);
        panic = TRUE;
      } else {
        act("$n skillfully retreats from battle", TRUE, this, 0, 0, TO_ROOM);
        panic = FALSE;
      }
      ```
      You're right Drew I didn't look at the code closely. But this code is pretty terrible. There's a for loop we iterate through 20 times with a variable i that isn't used for anything.
We determine what direction to try and then there's a skill chance later. What a mess

Reply:
So here's the dilemma we have with a lot of code. There's an undesired behavior or, as in this case, a bug. Fixing the bug improves gameplay experience but we're touching code that is terrible and should be refactored. Do we prioritize the gameplay experience and just merge the code or decide that someone that is touching it has the duty to improve the code?

I've tried to improve some code and I've left some shitty code that @cizra probably hates me for.

I feel bad asking you to refactor this and making your simple bug fix into a larger project @parkscs. I'll let you decide if you want to:
- merge this flee fix and let someone else worry about the code.
- pull it out of this PR and spend time to refactor.

Let me know.

Reply:
I actually already started completely refactoring that doFlee function on my own a couple days ago because retreat going the wrong way was annoying me so badly when I was testing thief stuff. So if you want, we can merge this PR just to get the retreat fix in and I'll just finish my refactor of flee that I've already started, @parkscs? That way you can keep going on your warrior stuff.